src/basic_hierarchy/interfaces/Hierarchy.java

```diff
@@ -1,12 +1,54 @@
 package basic_hierarchy.interfaces;
 
-public interface Hierarchy {
+/**
+ * A hierarchy is a collection of {@link Node}s, all of which share one common ancestor (the root group).
+ */
+public interface Hierarchy
+{
+	/**
+	 * Returns the root node of this hierarchy.
+	 */
 	public Node getRoot();
+
+	/**
+	 * Returns an array of groups of instances.
+	 */
 	public Node[] getGroups();
+
+	/**
+	 * Returns the number of groups in this hierarchy. Equivalent to {@code getGroups().length}.
+	 */
 	public int getNumberOfGroups();
+
+	/**
+	 * Returns an array of classes present in this hierarchy.
+	 */
 	public String[] getClasses();
+
+	/**
+	 * Returns the number of ground-truth classes. Equivalent to {@code getClasses().length}.
```

Comment:
@return the number of ground-truth classes. Equivalent to {@code getClasses().length}.